recipes/geogram/all/conanfile.py

```diff
@@ -0,0 +1,147 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+required_conan_version = ">=1.53.0"
+
+class GeogramConan(ConanFile):
+    name = "geogram"
+    description = "a programming library with geometric algorithms"
+    license = "BSD 3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/BrunoLevy/geogram"
+    topics = ("graphics programming", "mesh generation", "geometry processing", "graphics libraries", "mesh processing")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_graphics": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_graphics": False,
+    }
+
+    short_paths = True
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    # in case the project requires C++14/17/20/... the minimum compiler version should be listed
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_graphics:
+            self.requires("xorg/system")
+            self.requires("glfw/3.3.8")
+            self.requires("imgui/cci.20230105+1.89.2.docking")
+
+    def validate(self):
+        # validate the minimum cpp standard supported. For C++ projects only
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+        # in case it does not work in another configuration, it should validated here too
```

Comment:
please, pay attention when using the template to avoid using everything from there. This condition is an example, not a rule, most of projects should work shared + windows .